User sendpm bug

rica-carv opened the pull request:
### Motivation and Context
Enable passthrough of parms from the `USER_SENDPM` shortcode in user plugin to the `SENDPM` shortcode in e_shortcode
### Description
Corrected issue when parms where sent to the `USER_SENDPM` shortcode where ignored and never sent through to the `SENDPM` shortcode
### How Has This Been Tested?
Current github e107 code
### Types of Changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Performance enhancement (non-breaking change which improves efficiency)
- [x] Code cleanup (non-breaking change which makes code smaller or more readable)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (a change to man pages or other documentation)

### Checklist
- [x] My code adheres to the e107 [code style standard](https://github.com/e107inc/e107/wiki/e107-Coding-Standard).
- [x] I have read the [**contributing** document](https://github.com/e107inc/e107/blob/master/.github/CONTRIBUTING.md).
- [ ] I have added [tests](https://github.com/e107inc/e107/tree/master/e107_tests) to cover my changes.
- [ ] All new and existing tests pass.

rica-carv committed:
Merge branch 'e107inc:master' into user_sendpm-bug

rica-carv committed:
Update user_shortcodes.php

rica-carv committed:
Update user_shortcodes.php

CaMer0n commented on e107_core/shortcodes/batch/user_shortcodes.php:
```diff
@@ -462,11 +462,28 @@ function sc_user_forum_link($parm=null)
+            	)
+        		);
+			}
+
```
@rica-carv I believe this can be replaced with http_build_qry(); 

$parm = ['key1' => 'value1', 'key2' => 'value2'];
$query_string = http_build_query($parm);
// Output: key1=value1&key2=value2

rica-carv committed:
Update user_shortcodes.php

rica-carv committed:
Merge branch 'master' into user_sendpm-bug

rica-carv committed:
Merge branch 'e107inc:master' into user_sendpm-bug

rica-carv committed:
Merge branch 'e107inc:master' into user_sendpm-bug

rica-carv committed:
Merge branch 'e107inc:master' into user_sendpm-bug

rica-carv committed:
Merge branch 'e107inc:master' into user_sendpm-bug

rica-carv committed:
Merge branch 'e107inc:master' into user_sendpm-bug

rica-carv committed:
Merge branch 'e107inc:master' into user_sendpm-bug

e107help committed:
fix(user): resolve the parser and normalise the parm in sc_user_sendpm

$this->tp does not exist on user_shortcodes or on e_shortcode, so it fell
through e_shortcode::__get() into the shortcode variable bag and returned
null. The $tp added to the constructor was a local and went out of scope
immediately; the rest of this file takes a local parser per method, so
this does too.

$parm['user'] = ... also assumed an array. For the {USER_SENDPM=x} form
the parser hands over a string, and writing a named offset to a string
is a TypeError on PHP 8.

Forwarding the parms is the point of the change: {SENDPM} accepts 'class'
and 'glyph', and the old {SENDPM=id} form could only ever reach the id.
